assets/js/extensions/shipping-methods/pickup-location/utils.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { cleanForSlug } from '@wordpress/url';
+import { getSetting } from '@woocommerce/settings';
+
+/**
+ * Internal dependencies
+ */
+import type {
+	PickupLocation,
+	SortablePickupLocation,
+	ShippingMethodSettings,
+} from './types';
+
+export const indexLocationsById = (
+	locations: PickupLocation[]
+): SortablePickupLocation[] => {
+	return locations.map( ( value, index ) => {
+		return {
+			...value,
+			id: cleanForSlug( value.name ) + '-' + index,
+		};
+	} );
+};
+
+export const defaultSettings = {
+	enabled: 'yes',
+	title: '',
+	tax_status: 'taxable',
+	cost: '',
+};
+
+export const getInitialSettings = (): ShippingMethodSettings => {
+	const settings = getSetting(
+		'pickupLocationSettings',
+		defaultSettings
+	) as typeof defaultSettings;
+
+	return {
+		enabled: settings?.enabled === 'yes',
+		title: settings?.title || defaultSettings.title,
+		tax_status: settings?.tax_status || defaultSettings.tax_status,
+		cost: settings?.cost || defaultSettings.cost,
+	};
+};
+
+export const getInitialPickupLocations = (): SortablePickupLocation[] =>
+	indexLocationsById( getSetting( 'pickupLocations', [] ) );
```

Comment:
Settings might not be the best place to hydrate those things, you should feel free to just inline your data to a variable and pick that up.

Reply:
This is also not a blocker

Reply:
https://github.com/woocommerce/woocommerce-blocks/pull/7581/commits/1326cd615d54cb5ac18d2014420e8473256052d4